apps/aehttp/src/sc_ws_handler.erl

```diff
@@ -52,46 +85,125 @@ websocket_init(Params) ->
         {Handler, ChannelOpts} ->
             lager:debug("Starting Channel WS with params ~p",
                 [aesc_utils:censor_init_opts(Params)]),
-            lager:debug("ChannelOpts = ~p", [aesc_utils:censor_init_opts(ChannelOpts)]),
-            case start_link_fsm(Handler, ChannelOpts) of
-                {ok, FsmPid} ->
-                    MRef = erlang:monitor(process, FsmPid),
-                    {ok, Handler#handler{fsm_pid = FsmPid, fsm_mref = MRef}};
-                {error, Err} ->
-                    handler_init_error(Err, Handler)
-            end
+            websocket_init_parsed(Handler, ChannelOpts)
+    end.
+
+websocket_init_parsed(Handler, ChannelOpts) ->
+    lager:debug("ChannelOpts = ~p", [aesc_utils:censor_init_opts(ChannelOpts)]),
+    case start_link_fsm(Handler, ChannelOpts) of
+        {ok, FsmPid} ->
+            MRef = erlang:monitor(process, FsmPid),
+            {ok, Handler#handler{fsm_pid = FsmPid, fsm_mref = MRef}};
+        {error, Err} ->
+            handler_init_error(Err, Handler)
     end.
 
-websocket_init_reconnect(#{ <<"reconnect_tx">> := ReconnectTx } = Params) ->
+websocket_init_reconnect(#{ job_id := JobId
```

Comment:
I'd like to see documentation here explaining the chain of logic which is followed to try and reconnect.